lib/block-supports/settings.php

```diff
@@ -85,11 +85,14 @@ function _gutenberg_add_block_level_preset_styles( $pre_render, $block ) {
 	$registry                = WP_Block_Type_Registry::get_instance();
 	$blocks                  = $registry->get_all_registered();
 	foreach ( $blocks as $block_type ) {
-		if (
-			isset( $block_type->supports['__experimentalSelector'] ) &&
-			is_string( $block_type->supports['__experimentalSelector'] )
-		) {
-			$variables_root_selector .= ',' . $block_type->supports['__experimentalSelector'];
+		// We only want to append selectors for block's using custom selectors
```

Comment:
What would you think of doing the reverse? For each block, get its root selector, but only append it if it's different that `.wp-block-{block-name}`.

The checks we do here are also done within `wp_get_block_css_selector`. By embracing it, it hides the complexity of the selectors (legacy and new API) away. If any performance hit, we'll see it reported in the metrics.

Reply:
My first thoughts on this suggestion were that it would be an improvement. After taking a second look I'm not as sure.

The code as it stands assumes a block author isn't specifying `selectors.root` if it matches the standard `.wp-block-{name}` format. Given that is possible, it would be an improvement to omit adding selectors in that case.

It appears it's a choice between the possible inclusion of a selector if the block explicitly defines `selectors.root` as the standard class versus doing the generation of the default class twice for the majority of blocks. As you note, the ideal option might not be known until we see the resulting metrics.

I don't feel strongly about which way we go. Do you have a preference?

Given you've approved the current state of the PR, are you now leaning towards the original approach?

Reply:
I'm not too opinionated, specially given this doesn't impact the API. I go back and forth between the two. It's more explicit as it is, and that's always a good heuristic to follow in the absence of any other strong indicator.

Reply:
Alright, sounds like we can merge this and switch approaches later if we change our minds. Thanks 👍